src/Runner.Worker/ActionManager.cs

```diff
@@ -117,6 +131,22 @@ public async Task<List<JobExtensionRunner>> PrepareActionsAsync(IExecutionContex
                             imagesToBuildInfo[setupInfo.ActionRepository] = setupInfo;
                         }
                     }
+
+                    var repoAction = action.Reference as Pipelines.RepositoryPathReference;
+                    if (repoAction.RepositoryType != Pipelines.PipelineConstants.SelfAlias)
+                    {
+                        var definition = LoadAction(executionContext, action);
+                        if (definition.Data.Execution.HasInit)
```

Comment:
For container job, does pre run inside the container?

Reply:
yes, it will